pkg/cluster/acsk/action/action_utils.go

```diff
@@ -120,62 +120,207 @@ func attachInstancesToCluster(log logrus.FieldLogger, clusterID string, instance
 	return clusterWithPools, nil
 }
 
-func deleteNodepools(log logrus.FieldLogger, nodePools []*model.ACSKNodePoolModel, essClient *ess.Client, regionId string) error {
+func deleteNodePools(log logrus.FieldLogger, nodePools []*model.ACSKNodePoolModel, essClient *ess.Client, regionId string) error {
 	errChan := make(chan error, len(nodePools))
 	defer close(errChan)
 
 	for _, nodePool := range nodePools {
-		go func(nodePool *model.ACSKNodePoolModel) {
-
-			deleteSGRequest := ess.CreateDeleteScalingGroupRequest()
-			deleteSGRequest.SetScheme(requests.HTTPS)
-			deleteSGRequest.SetDomain(fmt.Sprintf(acsk.AlibabaESSEndPointFmt, regionId))
-			deleteSGRequest.SetContentType(requests.Json)
-			if nodePool.AsgID == "" {
-				// Asg could not be created nothing to remove
+		// TODO: run node pool deletion in parallel once Alibaba ESS API permits running multiple DeleteScalingGroup requests in parallel
+		// TODO: Currently running DeleteScalingGroup requests in parallel may fail with throttling error
+		deleteNodePool(log, nodePool, essClient, regionId, errChan)
+	}
+	var err error
+	caughtErrors := emperror.NewMultiErrorBuilder()
+
+	for i := 0; i < len(nodePools); i++ {
+		err = <-errChan
+		if err != nil {
+			caughtErrors.Add(err)
+		}
+	}
+
+	return pkgErrors.NewMultiErrorWithFormatter(caughtErrors.ErrOrNil())
+}
+
+func deleteNodePool(log logrus.FieldLogger, nodePool *model.ACSKNodePoolModel, essClient *ess.Client, regionId string, errChan chan<- error) {
+	deleteSGRequest := ess.CreateDeleteScalingGroupRequest()
+	deleteSGRequest.SetScheme(requests.HTTPS)
+	deleteSGRequest.SetDomain(fmt.Sprintf(acsk.AlibabaESSEndPointFmt, regionId))
+	deleteSGRequest.SetContentType(requests.Json)
+	if nodePool.AsgID == "" {
+		// Asg could not be created nothing to remove
+		errChan <- nil
+		return
+	}
+
+	deleteSGRequest.ScalingGroupId = nodePool.AsgID
+	deleteSGRequest.ForceDelete = requests.NewBoolean(true)
+
+	_, err := essClient.DeleteScalingGroup(deleteSGRequest)
+	if err != nil {
+		if sdkErr, ok := err.(*aliErrors.ServerError); ok {
+			if strings.Contains(sdkErr.ErrorCode(), "InvalidScalingGroupId.NotFound") {
+				log.WithFields(logrus.Fields{"scalingGroupId": nodePool.AsgID, "nodePoolName": nodePool.Name}).Info("scaling group to be deleted not found")
+
 				errChan <- nil
 				return
 			}
+		}
 
-			deleteSGRequest.ScalingGroupId = nodePool.AsgID
-			deleteSGRequest.ForceDelete = requests.NewBoolean(true)
+		errChan <- emperror.WrapWith(err, "could not delete scaling group", "scalingGroupId", nodePool.AsgID, "nodePoolName", nodePool.Name)
+		return
+	}
 
-			_, err := essClient.DeleteScalingGroup(deleteSGRequest)
-			if err != nil {
+	_, err = waitUntilScalingInstanceCreated(log, essClient, regionId, nodePool)
```

Comment:
I'd update the function's name to
```suggestion
	_, err = waitUntilScalingInstancesUpdated(log, essClient, regionId, nodePool)
```
to reflect its purpose better